src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -83,27 +83,27 @@ float64_t RandomFourierDotFeatures::post_dot(float64_t dot_result, index_t par_i
 	return std::cos(dot_result) * constant;
 }
 
-SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
+void RandomFourierDotFeatures::generate_random_parameter_vector(SGMatrix<float64_t>& mat, index_t col)
 {
 	NormalDistribution<float64_t> normal_dist;
 	UniformRealDistribution<float64_t> uniform_real_dist(0.0, 2 * Math::PI);
-	SGVector<float64_t> vec(feats->get_dim_feature_space()+1);
+	auto len = mat.num_rows + 1;       
```

Comment:
it is a bad idea to rely on the input for the size. Rather assert/require that the input (vector) has dimension of feature space + 1 as size.